pom.xml

```diff
@@ -200,225 +249,305 @@
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <!-- JetBrains Java Annotations -->
+            <groupId>org.jetbrains</groupId>
+            <artifactId>annotations</artifactId>
+            <version>21.0.1</version>
+            <type>jar</type>
+            <scope>compile</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <!-- Spigot-API -->
             <groupId>org.spigotmc</groupId>
             <artifactId>spigot-api</artifactId>
             <version>1.17-R0.1-SNAPSHOT</version>
             <type>jar</type>
-            <scope>compile</scope>
+            <scope>provided</scope>
+            <optional>true</optional>
         </dependency>
         <dependency>
+            <!-- Spigot-Server -->
+            <groupId>org.spigotmc</groupId>
+            <artifactId>spigot</artifactId>
+            <version>1.17-R0.1-SNAPSHOT</version>
+            <type>jar</type>
+            <scope>provided</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <!-- PaperLib -->
             <groupId>io.papermc</groupId>
             <artifactId>paperlib</artifactId>
-            <version>1.0.6</version>
+            <version>1.0.6-SNAPSHOT</version>
+            <type>jar</type>
+            <scope>compile</scope>
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <!-- Floodgate -->
+            <groupId>org.geysermc.floodgate</groupId>
+            <artifactId>api</artifactId>
+            <version>2.0-SNAPSHOT</version>
+            <type>jar</type>
             <scope>compile</scope>
+            <optional>true</optional>
         </dependency>
         <dependency>
+            <!-- TARDISChunkGenerator -->
             <groupId>me.eccentric_nz.tardischunkgenerator</groupId>
             <artifactId>TARDISChunkGenerator</artifactId>
-            <version>4.7.3</version>
+            <version>4.7.3-SNAPSHOT</version>
             <type>jar</type>
             <scope>compile</scope>
+            <optional>true</optional>
         </dependency>
         <dependency>
+            <!-- TARDISWeepingAngels -->
             <groupId>me.eccentric_nz.tardisweepingangels</groupId>
             <artifactId>TARDISWeepingAngels</artifactId>
-            <version>3.3.3</version>
+            <version>3.4.0-SNAPSHOT</version>
             <type>jar</type>
             <scope>compile</scope>
+            <optional>true</optional>
         </dependency>
         <dependency>
-            <groupId>com.sk89q</groupId>
-            <artifactId>worldguard</artifactId>
-            <version>7.0.5</version>
+            <!-- WorldEdit -->
+            <groupId>com.sk89q.worldedit</groupId>
+            <artifactId>worldedit-bukkit</artifactId>
+            <version>7.3.0-SNAPSHOT</version>
             <type>jar</type>
-            <scope>system</scope>
```

Comment:
I know this is wrong and Maven complains about it, but in the past some repositories just disappear or aren't reachable and the build fails. Until the feature actually becomes unsupported, I'd rather stick with it - unless I'm doing something else completely wrong, I'm happy to be informed so :)

Reply:
I can understand this fear for one-off plugins and repos, but for repos that are reliant upon by many and have a sort of level of trustworthiness (the enginehub plugins, essentials, to name a few) should be handled in the pom here.
The absolute worst-case scenario is to handle and install these locally onto your Jenkins (most of the time the source is open, so shouldn't be an issue?) then compile and build as needed.

Reply:
Totally agree, but I think I started doing this for Multiverse years back, and I would consider that plugin to be fairly reliable... it just wasn't at the time.

Reply:
This would also mean that maven will start downloading dependencies of dependencies (at least the first time) and the build time is forever and my local .m2 directory is full of crap I neither want or need.

Reply:
I've just discovered `<exclusions>`... but build times are still pretty slow as were checking so many different repos...